client/src/components/WinLoss.svelte

```diff
@@ -0,0 +1,202 @@
+<script lang="ts">
+  import type { PrivatePlayer, Lobby, Role } from "../../../functions/src/firestore-types/lobby";
+  import { lobbyReturn } from "$lib/firebase/firebase-functions";
+  import { authStore as user } from "$stores/auth";
+
+  export let lobbyCode: string;
+  export let lobby: Lobby;
+  export let privatePlayer: PrivatePlayer;
+
+  let end: "Cat Win" | "Cat Lose" | "Catfish Lose" | "Catfish Win" | null = null;
+  let catfishList: string = "";
+  let catfishes: string[];
+  let isHost: boolean;
+  let myRole: Role;
+
+  // check if the user is the host
+  $: if ($user !== null) {
+    if (lobby.uids.indexOf($user.uid) == 0) {
+      isHost = true;
+    } else {
+      isHost = false;
+    }
+  }
+  // format a string list to have commas and "and" when needed
+  const formatter = new Intl.ListFormat("en", { style: "long", type: "conjunction" });
+  // get the list of catfishes from lobby.ts and put them into a variable
+  catfishes = lobby.players
+    .filter((p) => {
+      return p.role == "CATFISH";
+    })
+    .map((c) => c.displayName);
+  catfishList = formatter.format(catfishes);
+
+  // grab the current user's role
+  myRole = privatePlayer.role;
```

Comment:
this is not reactive but honestly it would probobly be better to get rid of myRole entirly and just use privatePlayer.role on line 39